compiler/daml-lf-proto/src/DA/Daml/LF/Proto3/EncodeV1.hs

```diff
@@ -640,9 +656,9 @@ encodeFeatureFlags FeatureFlags{..} = Just P.FeatureFlags
 
 encodeModuleWithLargePackageIds :: Version -> Module -> P.Module
 encodeModuleWithLargePackageIds version mod =
-    let env = EncodeEnv version S.empty
+    let env = initEncodeEnv version InternNothing
```

Comment:
Have we ever checked if this is actually faster? Due to the increase in allocations I wouldn’t be surprised if this actually ends up being slower in which case there is no point keeping this for the scenario service.

Reply:
The problem right now is that the scenario service takes a single module rather than a package. The string interning table is part of the package not of the module. We should change this by shipping a "fat module", i.e. a module with a string interning table to the scenario service. That is on my todo list.

Reply:
Oh I see, that makes a lot of sense.